orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/providers/kubernetes/KubernetesJobRunner.java

```diff
@@ -48,6 +56,23 @@ public KubernetesJobRunner(ArtifactResolver artifactResolver, ObjectMapper objec
       operation.putAll(stage.getContext());
     }
 
+    // previous versions of run job did not support artifacts so we'll default to text here
+    String source = (String) stage.getContext().getOrDefault("source", "text");
```

Comment:
Would it make sense to move the `stage.mapTo(RunJobManifestContext.class)` up here and then just use `runJobManifestContext.getSource()` to avoid the casting and string matching?